hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4815,3 +4878,211 @@ func isAPIServerRoute(hcluster *hyperv1.HostedCluster) bool {
+									},
+								},
+							},
+							{
+								Name: "HOSTED_CONTROL_PLANE_NAMESPACE",
+								ValueFrom: &corev1.EnvVarSource{
+									FieldRef: &corev1.ObjectFieldSelector{
+										FieldPath: "metadata.namespace",
+									},
+								},
+							},
+							{
+								Name:  "HOSTED_CONTROL_PLANE_NAME",
+								Value: hcp.Name,
+							},
+							{
+								Name:  "CERT_ROTATION_SCALE",
+								Value: certRotationScale.String(),
+							},
+						},
+						Command:   []string{"/usr/bin/control-plane-pki-operator"},
+						Args:      []string{"operator"},
+						Ports:     []corev1.ContainerPort{{Name: "metrics", Protocol: "TCP", ContainerPort: 8443}},
+						Resources: cpoResources,
+					},
+				},
+			},
+		},
+	}
+
+	if openShiftTrustedCABundleConfigMapExists {
+		hyperutil.DeploymentAddOpenShiftTrustedCABundleConfigMap(deployment)
+	}
+
+	mainContainer = hyperutil.FindContainer("control-plane-pki-operator", deployment.Spec.Template.Spec.Containers)
+	proxy.SetEnvVars(&mainContainer.Env)
+
+	// set security context
+	if setDefaultSecurityContext {
+		deployment.Spec.Template.Spec.SecurityContext = &corev1.PodSecurityContext{
+			RunAsUser: k8sutilspointer.Int64(config.DefaultSecurityContextUser),
+		}
+	}
+
+	deploymentConfig := config.DeploymentConfig{
+		Scheduling: config.Scheduling{
+			PriorityClass: config.DefaultPriorityClass,
+		},
+		SetDefaultSecurityContext: setDefaultSecurityContext,
+		AdditionalLabels: map[string]string{
+			config.NeedManagementKASAccessLabel: "true",
+		},
+	}
+	if hcp.Annotations[hyperv1.ControlPlanePriorityClass] != "" {
+		deploymentConfig.Scheduling.PriorityClass = hcp.Annotations[hyperv1.ControlPlanePriorityClass]
+	}
+	deploymentConfig.SetDefaults(hcp, nil, k8sutilspointer.Int(1))
+	deploymentConfig.SetRestartAnnotation(hc.ObjectMeta)
+	deploymentConfig.ApplyTo(deployment)
+
+	return nil
+}
+
+func reconcileControlPlanePKIOperatorRole(role *rbacv1.Role) error {
+	role.Rules = []rbacv1.PolicyRule{
+		{ // to create owner refs
+			APIGroups: []string{"hypershift.openshift.io"},
+			Resources: []string{"hostedcontrolplanes"},
+			Verbs:     []string{"get"},
+		},
+		{ // to report status
+			APIGroups: []string{"hypershift.openshift.io"},
+			Resources: []string{"hostedcontrolplanes/status"},
+			Verbs:     []string{"patch"},
+		},
+		{ // to do the work of the controller
+			APIGroups: []string{""},
+			Resources: []string{"configmaps", "secrets", "events"},
+			Verbs:     []string{"get", "list", "watch", "create", "delete", "update", "patch"},
+		},
+		{ // for owner reference resolution
+			APIGroups: []string{""},
+			Resources: []string{"pods"},
+			Verbs:     []string{"get"},
+		},
+		{ // for owner reference resolution
+			APIGroups: []string{"apps"},
+			Resources: []string{"replicasets"},
+			Verbs:     []string{"get"},
+		},
+		{ // for owner reference resolution
+			APIGroups: []string{"config.openshift.io"},
+			Resources: []string{"infrastructures"},
```

Comment:
I wouldn't think this works in a role

Reply:
Oops, forgot to remove here -it's in the cluster role below